src/gennylib/include/gennylib/parallel.hpp

```diff
@@ -78,17 +116,16 @@ class ExceptionBucket {
         }
 
     private:
-        std::deque<std::exception_ptr> _caughtExceptions;
+        std::vector<std::exception_ptr> _caughtExceptions;
```

Comment:
is `exception_ptr` really the right type to be using here? I think it's a global singleton pointing to what's returned by `current_exception` so the underlying objects may have gone away. Perhaps this wants to be a `std::vector<std::string> _whatMessages` or similar?

Reply:
The exception_ptr is a shared-ptr-like type that promises that the exception it manages exists as long as it does. That has its own problems, of course, but I think should be fine as long as this exception bucket and the exception thrown from it don't hang around forever?